Vineyard/Vineyard/CreateGroupView.swift

```diff
@@ -51,8 +53,16 @@ struct CreateGroupView: View {
                 .padding([.leading, .trailing], 10)
                 
                 Spacer()
-                NavigationLink {
-                    GoalsListView(groupName: $groupName, resolution: $resolution, deadline: $deadline)
+                Button {
+                    
+                    do {
```

Comment:
All this logic, including the state variables, should be moved to the view model.

Reply:
alright I moved the items